### PR TITLE
test(x/data): update iri tests and remove duplicates

### PR DESCRIPTION
**Pull request by ryanchristo**
## Description

Closes: #639

Add test cases for `ToIRI` and `ParseIRI` and remove duplicate test cases handled in `types_test.go`.

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/data/iri.go [hunk 1]
```diff
@@ -120,24 +120,24 @@ func ParseIRI(iri string) (*ContentHash, error) {
 	const regenPrefix = "regen:"
 
 	if !strings.HasPrefix(iri, regenPrefix) {
-		return nil, ErrInvalidIRI.Wrapf("can't parse IRI %s without %s prefix", iri, regenPrefix)
+		return nil, ErrInvalidIRI.Wrapf("failed to parse IRI %s, %s prefix required", iri, regenPrefix)
```
Comment (technicallyty) [inline x/data/iri.go:123]:
```suggestion
		return nil, ErrInvalidIRI.Wrapf("failed to parse IRI %s: %s prefix required", iri, regenPrefix)
```

nit: imo error messages look a bit better when the reason is trailed by a `:` rather than `,`

**Review by technicallyty — APPROVED** [review]
lgtm, 1 nit

File: x/data/iri.go (hunk 1 above)
Comment (ryanchristo) [reply x/data/iri.go]: I updated to use `:` except for when the error includes `, expected`, which I think a `,` still makes sense. https://github.com/regen-network/regen-ledger/pull/950/commits/a16c57abf82937abe9b194f73ce8632e508503a2

**Review by aleem1314 — APPROVED** [review]
lgtm
